modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java

```diff
@@ -41,10 +42,15 @@
 
     protected ReadWriteLock lockDetails = new ReentrantReadWriteLock();
 
+    /** local transformed code cache */
+    protected Map<AionAddress, TransformedCodeInfo> cachedTransformedCode;
+    protected ReadWriteLock lockTransformedCode = new ReentrantReadWriteLock();
```

Comment:
Please remove this lock and use the full write lock method (for both read and write). Introducing another lock is too risky here, e.g. `updateBatch` is missing the necessary lock. I will open an issue to reduce locking to a single object in this class. The current use of multiple locks is risky due to the write required by loading the account state and contract details and not particularly useful since the cache is rarely accessed by multiple threads.